src/hdmf/common/table.py

```diff
@@ -468,6 +474,10 @@ def __init__(self, **kwargs):  # noqa: C901
         self.__colids = {name: i + 1 for i, name in enumerate(self.colnames)}
         self._init_class_columns()
 
+        if target_tables:
+            self._set_dtr_targets(target_tables)
```

Comment:
Since a user may specify the DTR via the `columns` argument as well, do we need to check that the `target_table` and `columns` are consistent, i.e., what if the target specified in columns is different from what is listed in target_tables?

Reply:
Yes, that is checked on line 558.